tests/mellowLpRouter.test.ts

```diff
@@ -614,9 +616,149 @@ describe('Mellow Router Test Suite', () => {
       await ethMellowLpRouter.deposit(2, [0, 100]);
       await ethMellowLpRouter.deposit(3, [50, 50]);
 
-      expect(ethMellowLpRouter.userDeposit).to.be.eq(1);
+      expect(ethMellowLpRouter.userComittedDeposit).to.be.eq(1);
       expect(ethMellowLpRouter.userPendingDeposit).to.be.eq(5);
-      expect(ethMellowLpRouter.userTotalDeposit).to.be.eq(6);
+      expect(ethMellowLpRouter.userDeposit).to.be.eq(6);
+    });
+  });
+
+  describe('Withdrawal Scenarios', async () => {
+    beforeEach('Setting up the Router Object', async () => {
+      await resetNetwork(8099085);
+      await extendRouter();
+
+      ethMellowLpRouter = new MellowLpRouter({
+        mellowRouterAddress: MellowRouterAddress,
+        defaultWeights: [50, 50],
+        provider,
+      });
+
+      await ethMellowLpRouter.vaultInit();
+
+      // Initialise the user so the router contract is connected with user to keep track of them as a signer for the deposits
+      await ethMellowLpRouter.userInit(userWallet);
+    });
+
+    it('2-vault withdrawal', async () => {
+      const wethContract = new ethers.Contract(
+        ethMellowLpRouter.readOnlyContracts?.token.address || '',
+        IERC20MinimalABI,
+        signer,
+      );
+
+      // User deposits funds into the router
+      await ethMellowLpRouter.deposit(10);
+
+      // Submit the batch of deposits from the router to the erc20 root vaults
+      for (let vaultIndex = 0; vaultIndex < 2; vaultIndex += 1) {
+        await ethMellowLpRouter.writeContracts?.mellowRouter.submitBatch(vaultIndex, 0);
+      }
+
+      await advanceTimeAndBlock(30 * 24 * 60 * 60, 1);
+
+      await ethMellowLpRouter.refreshUserDeposit();
+      expect(ethMellowLpRouter.userDeposit).to.be.eq(10);
+
+      const initBalance: BigNumber = await wethContract.balanceOf(ethMellowLpRouter.userAddress);
+
+      await ethMellowLpRouter.withdraw(0);
+
+      await ethMellowLpRouter.refreshUserDeposit();
```

Comment:
it relates to my earlier comment: shouldn't we refresh user deposit within the withdraw function? (e.g., we can have it as`refreshUserDeposit(refreshCommitted=true/false, refreshPending=true/false)`)

Reply:
That's true but I'd keep it general for now.